### PR TITLE
Directory

### PR DESCRIPTION
**Pull request by juchen3637**

### CODE REVIEW

File: src/server/api/routers/user.ts [hunk 1]
```diff
@@ -26,4 +26,15 @@ export const userRouter = createTRPCRouter({
   getAll: protectedProcedure.query(async () => {
     return await db.user.findMany();
   }),
+
+  getByRole: protectedProcedure.input(z.string()).query(async ({ input }) => {
```
Comment (ulrokx) [inline src/server/api/routers/user.ts:30]: It's a bit of a nit, but could we make this input an object with one field named role? Single object argument functions are pretty common with JS/TS and this function might want to take more arguments in the future (search, offset, limit)

File: src/server/api/routers/user.ts [hunk 2]
```diff
@@ -26,4 +26,15 @@ export const userRouter = createTRPCRouter({
...
+    return await db.profile.findMany({
```
Comment (ulrokx) [inline src/server/api/routers/user.ts:31]: Async functions return promises and tRPC will await any promise we return, so we don't need to await the promise returned by Prisma here before returning in this function. The behavior will be identical for both ways, with or without the await here, but we don't need the await so better to leave it out.

Thinking about it, depending on the JS runtime implementation, awaiting here might cost more memory as an intermediate promise object might be made here and resolved before returning. Either way you can drop the await and it still works. :D

File: src/pages/directory.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { api } from "~/utils/api";
+import { Button, ButtonGroup, Stack } from "@mui/material";
+import { useState } from "react";
+import Profile from "../components/directory/Profile";
+
+const Directory = () => {
+  const [current, setCurrent] = useState<string>("mentee");
```
Comment (ulrokx) [inline src/pages/directory.tsx:7]: We can use[ Typescript enums](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) to our advantage here. You can make the enum names all caps and make the strings the way you want to display it. You can use toUpperCase for the argument to useQuery.

https://stackoverflow.com/a/74326516

File: src/pages/directory.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      onClick={() => setCurrent(role.toLowerCase())}
+    >
+      {role}
+    </Button>
+  ));
+
+  return (
+    <main className="flex min-h-screen w-full flex-col items-center bg-clear">
+      <h1 className="mt-6 text-6xl font-bold text-aero">Directory</h1>
+      <ButtonGroup className="mb-4 mt-10" size="large" color="primary">
```
Comment (ulrokx) [inline src/pages/directory.tsx:25]: If these are tabs and there is a currently selected tab, maybe [this ToggleButton component](https://mui.com/material-ui/react-toggle-button/) is a better fit here?

**Review by ulrokx — CHANGES_REQUESTED** [review]
Nice job

**Comment by miguel-merlin** [comment]
@ulrokx I didn't have time to look into the type inference with tRPC. Can we discuss it tomorrow? Also, build is failing because of problems with ```cp .env.example```

**Comment by miguel-merlin** [comment]
@ulrokx Ready for review!

**Comment by miguel-merlin** [comment]
#89

**Review by ShawnAviles — APPROVED** [review]
Looks good as long as @ulrokx approves of the follow-up changes. I have a PR coming that will need to edit some of this code (some new UI / Styling changes from working on #126); very minimal but hope to have this merged soon

File: src/utils/useProtectedPage.ts [hunk 5]
```diff
@@ -12,5 +12,5 @@ export default function useProtectedPage(roles: RoleName[]) {
         pathname: "/",
       });
     }
-  }, [session, router]);
+  }, [session, router, roles]);
```
Comment (ulrokx) [inline src/utils/useProtectedPage.ts:15]: Nice

**Review by ulrokx — APPROVED** [review]
👍

**Pull request merged**